# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/DrawTargetWebglInternal.h
# Commit: 3d56b847ddf9
# Full Hash: 3d56b847ddf9ff602f274fbcbaa34421e925b3a3
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2022-02-12 09:47:43
# Regressor Bug: 1754130
# File Overlap Count: 2
# Description:
#   Bug 1754130 - Implement SharedContext for sharing WebGL state among many DrawTargetWebgls. r=aosmond
#   
#   Within DrawTargetWebgl, this implements a SharedContext class to store both the WebGL context and
#   also related resources for drawing with it. Not a lot of new code has actually been added as this
#   was mostly accomplished by refactoring as much drawing code as possible that touched the WebGL
# ==============================================================================

diff -r 084ec5de6c39 -r 3d56b847ddf9 dom/canvas/DrawTargetWebglInternal.h
--- a/dom/canvas/DrawTargetWebglInternal.h	Fri Feb 11 15:19:47 2022 +0000
+++ b/dom/canvas/DrawTargetWebglInternal.h	Fri Feb 11 15:36:29 2022 +0000
@@ -147,7 +147,7 @@
 
   virtual void UpdateSize(const IntSize& aSize) {}
 
-  virtual void Cleanup(DrawTargetWebgl& aDT) {}
+  virtual void Cleanup(DrawTargetWebgl::SharedContext& aContext) {}
 
   virtual ~TextureHandle() {}
 
@@ -248,7 +248,7 @@
     return TextureHandle::UsedBytes(GetFormat(), mBounds.Size());
   }
 
-  void Cleanup(DrawTargetWebgl& aDT) override;
+  void Cleanup(DrawTargetWebgl::SharedContext& aContext) override;
 
   const RefPtr<SharedTexture>& GetOwner() const { return mTexture; }
 
@@ -284,7 +284,7 @@
 
   void UpdateSize(const IntSize& aSize) override { mSize = aSize; }
 
-  void Cleanup(DrawTargetWebgl& aDT) override;
+  void Cleanup(DrawTargetWebgl::SharedContext& aContext) override;
 
  private:
   IntSize mSize;